src/services/pull/usePullBADropdownSummary.tsx

```diff
@@ -0,0 +1,170 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import {
+  FirstPullRequestSchema,
+  MissingBaseCommitSchema,
+  MissingBaseReportSchema,
+  MissingComparisonSchema,
+  MissingHeadCommitSchema,
+  MissingHeadReportSchema,
+} from 'services/comparison/schemas'
+import {
+  RepoNotFoundErrorSchema,
+  RepoOwnerNotActivatedErrorSchema,
+} from 'services/repo/schemas'
+import Api from 'shared/api'
+import A from 'ui/A'
+
+const BAComparisonSchema = z.object({
+  __typename: z.literal('BundleAnalysisComparison'),
+  sizeDelta: z.number(),
+  loadTimeDelta: z.number(),
+})
+
+const BundleAnalysisCompareWithParentSchema = z
+  .discriminatedUnion('__typename', [
+    BAComparisonSchema,
+    FirstPullRequestSchema,
+    MissingBaseCommitSchema,
+    MissingBaseReportSchema,
+    MissingComparisonSchema,
+    MissingHeadCommitSchema,
+    MissingHeadReportSchema,
+  ])
+  .nullable()
+
+const RepositorySchema = z.object({
+  __typename: z.literal('Repository'),
+  pull: z
+    .object({
+      bundleAnalysisCompareWithBase:
+        BundleAnalysisCompareWithParentSchema.nullable(),
+    })
+    .nullable(),
+})
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      repository: z.discriminatedUnion('__typename', [
+        RepositorySchema,
+        RepoNotFoundErrorSchema,
+        RepoOwnerNotActivatedErrorSchema,
+      ]),
+    })
+    .nullable(),
+})
+
+const query = `
+query PullBADropdownSummary(
+  $owner: String!
+  $repo: String!
+  $pullId: Int!
+) {
+  owner(username: $owner) {
+    repository(name: $repo) {
+      __typename
+      ... on Repository {
+        pull(id: $pullId) {
+          bundleAnalysisCompareWithParent {
+            __typename
+            ... on BundleAnalysisComparison {
+              sizeDelta
+              loadTimeDelta
+            }
+            ... on FirstPullRequest {
+              message
+            }
+            ... on MissingBaseCommit {
+              message
+            }
+            ... on MissingHeadCommit {
+              message
+            }
+            ... on MissingComparison {
+              message
+            }
+            ... on MissingBaseReport {
+              message
+            }
+            ... on MissingHeadReport {
+              message
+            }
+          }
+        }
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}`
+
+interface UseCommitDropdownSummaryArgs {
+  provider: string
+  owner: string
+  repo: string
+  pullId: number
+}
+
+export function usePullBADropdownSummary({
+  provider,
+  owner,
+  repo,
+  pullId,
+}: UseCommitDropdownSummaryArgs) {
+  return useQuery({
+    queryKey: ['CommitSummary', provider, owner, repo, pullId],
```

Comment:
Think we want `PullBADropdownSummary`. 

I might have missed this in the previous PR so that might need an update.

Reply:
Nice catch, all fixed up.